### DISCUSSION TITLE
Issues reading SPC file contents into Dash App with scp.read_spc(content=content)

### DISCUSSION THREAD

**Discussion by seand412** [spectrochempy/spectrochempy#570] [Q&A] [answered]
Hi, I'm trying to create a Dash app which requires reading of SPC spectral files. Due to the quirks of Dash, I can't pass a file location, I have to read in the file contents. SpectroChemPy docs for the `scp.read_spc()` function (and other reads) state I can use the `content` kwarg to pass a bytes object instead. When I try (in a Jupyter notebook) though, I get an error message that states "ERROR | TypeError: 'WindowsPath' object is not subscriptable." in a red error box under the executed cell. There's no traceback provided. 

The docs state there are examples on how to do what I'm trying in the `tests/test_readers` directory, but in my environment I can't find this directory. And the `test_read_spc.py` file that I found in an older version online just has a bunch of Assert tests to ensure files were read properly, not any examples on how to read spc file contents.

The file I'm trying to read is an SPC file, I read the bytes contents of it with `with open(file, 'rb') as f: content = f.read()`. I try reading the contents with `f = scp.read_spc(content=content)`. Then I get the TypeError issue.

Any help would be much appreciated, let me know if you need more info and I'll see what I can do.

Thanks!

**Comment by fernandezc** [comment, marked as answer]
see the discussion on issue #571 for the solution to this problem.
